### PR TITLE
Update instruction macros to use .insn directive

### PR DESCRIPTION
**Pull request by pranav-prakash**
The prior solution relied on deprecated register storage class specifiers.
Note that .insn directives require a sufficiently recent version of binutils.

Addresses issue #3.

DCO 1.1 Signed-off-by: Pranav Prakash <pranav-prakash@users.noreply.github.com>

### CODE REVIEW

File: src/xcustom.h [hunk 1]
```diff
@@ -19,62 +19,32 @@
...
-  XCUSTOM_OPCODE(x)                   |         \
-  (rd                   << (7))       |         \
-  (0x3                  << (7+5))     |         \
-  ((rd != 0) & 1        << (7+5+2))   |         \
-  (rs1                  << (7+5+3))   |         \
-  (rs2                  << (7+5+3+5)) |         \
-  (EXTRACT(funct, 7, 0) << (7+5+3+5+5))
-
-#define ROCC_INSTRUCTION_RAW_R_R_R(x, rd, rs1, rs2, funct)      \
-  .word XCUSTOM(x, ## rd, ## rs1, ## rs2, funct)
```
Comment (seldridge) [inline src/xcustom.h:51]: Can the PR leave this in? 

I still use this for some straight riscv-test-style assembly language tests where I'm actually writing assembly as opposed to inline assembly stuck into a C program.

**Review by seldridge — CHANGES_REQUESTED** [review]
This is awesome. Thanks for taking the time to do what I didn't do for this.

I was playing around with this locally and had some modifications that I'll PR against this PR now.

File: src/xcustom.h (hunk 1 above)
Comment (pranav-prakash) [reply src/xcustom.h:51]: Seems like this has been addressed this in the PR you PR'd against my PR.

**Comment by pranav-prakash** [comment]
I merged the changes you PR'd against my fork so I believe everything should be good to go.

I've been testing the C macros against proxy kernel since I'm using this for gemmini (née systolic) and it works fine there. And you mentioned you tested out bare metal so that is hopefully thorough enough.

**Review by seldridge — APPROVED** [review]

**Comment by seldridge** [comment]
This is great! Thanks for getting this together.

**Pull request merged**